torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2921,6 +2921,16 @@ def issue_guard(guard: ShapeGuard) -> None:
                 if len(bounds) > 1:
                     exprs.append(" <= ".join(bounds))
 
+        # 4. Add divisibility guards.
+        for mod_expr in self.divisible:
+            guard_expr = sympy.Eq(self.replace(mod_expr), sympy.Integer(0))
+            exprs.append(
+                ShapeGuardPrinter(symbol_to_source, source_ref, self.var_to_sources).doprint(guard_expr)
+            )
+            self._add_target_expr(guard_expr)
+            if any(is_dim(source) for s in mod_expr.free_symbols for source in symbol_to_source[s]):
+                self.dim_constraints.add(guard_expr)
```

Comment:
Previously, we would omit divisibility guards entirely -- even though they were in `self.guards` -- because they would be simplified away into constants (since `issue_guard` calls `simplify()`). I'm not certain that recreating these guards from `self.divisible` is the right way to go, though; maybe I should ensure divisibility guards get correctly handled by `issue_guard`? There's a bunch of logic here that's copypasted from `issue_guards` after all.